### ISSUE TITLE
Explicit Conditional Constants added to Project file Duplicate Implicit ones

### ISSUE THREAD

**Issue by natidea** [dotnet/project-system#2720]
Based on [VSFeedback](https://developercommunity.visualstudio.com/content/problem/73187/conditional-compilation-symbols-are-duplicated-on.html?childToView=95513#comment-95513) | [457114](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/457114)

When a custom conditional compilation constant is added to a project, it inserts an explicit constant for the framework, which duplicates the implicit ones.

**Repro Steps**
- Create new .NET Core Console App
- Open Properties | Build and add a constant to Conditional Compilation symbols:
![image](https://user-images.githubusercontent.com/7732033/29389702-9575eada-82a0-11e7-8f5e-ce409843855c.png)
- Close and Reopen Property pages

**Expected**
Constants remain the same

**Actual**
NETCOREAPP1_1 is repeated twice:
![image](https://user-images.githubusercontent.com/7732033/29389741-d037c076-82a0-11e7-98de-907013247fac.png)

**Notes and Workaround**
An explicit constant is added to the project file which duplicates the implicit constant:
```xml
  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
    <DefineConstants>TRACE;DEBUG;NETCOREAPP1_1;SOMECONSTANT</DefineConstants>
  </PropertyGroup>
```
Removing that constant eliminates the duplicate in the property pages, and it is likely not needed because an implicit constant is defined at [Microsoft.NET.Sdk.CSharp.targets#L34](https://github.com/dotnet/sdk/blob/64a352d4b4b99509e0a513584afef0ac3292fa34/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.CSharp.targets#L34)

This is likely the root cause of #1440 which can be duped against this.

**Comment by rolandh** [comment]
How is removing the constant a fix for this?

I have my own build symbols I want to add and every time I resave the project it adds NET461 etc again. What is the work around for this?

**Comment by ericbrunner** [comment]
Totally agree with @rolandh. See my comments on that in context of issue https://github.com/dotnet/project-system/issues/2733#issuecomment-352244818
When the hell is that fixed? I work with latest stable VS2017 15.5.2 from 14th December and it is still not fixed.

**Comment by rolandh** [comment]
I stopped using netcore literally because of how annoying this was.

**Comment by rolandh** [comment]
The whole. New csproj format seems flaky and unfinished. I do not feel that netcore is production ready.

**Comment by davkean** [comment]
@rolandh Above is not suggesting removing the constant as a fix for this, it's providing a workaround for those affected by it.

Understood how frustrating this bug is - we've been prioritizing bugs that affect the inner loop (things developers do over and over again) and with no workaround, over issues are affect outer loop (things developers only do a few times a day or less) with a workaround.

**Comment by jmroyb** [comment]
Still happening in v15.6.4

**Comment by varungarg2210** [comment]
It seems quite old now and is yet not fixed till today

**Comment by jmroyb** [comment]
Will be fixed in v15.6.7?

**Comment by jmroyb** [comment]
Still failing in v15.6.7. NetStandard project Scenarios failing:

Scenario 1
Set focus to "conditional compilation symbols" textbox and add semicolon and press Intro or TAB or Alt+Tab: textbox self modified with "NETSTANDARD2_0;NETSTANDARD2_0" text.

Scenario 2
Set focus to "conditional compilation symbols" textbox and add semicolon and other symbol and press Intro or TAB: textbox self modified with "NETSTANDARD2_0;mine;NETSTANDARD2_0". Not failing when press Alt+Tab combined keys.

Cheers

**Comment by Pilchie** [comment]
Reported again in :link: https://developercommunity.visualstudio.com/content/problem/242209/add-portable-to-conditional-compilation-and-the-te.html

**Comment by davkean** [comment]
I've submitted https://github.com/dotnet/project-system/pull/3579 to fix this.

**Comment by davkean** [comment]
This will be fixed in 15.8 Preview 3.

**Comment by jmroyb** [comment]
Seems solved in 15.8.

Cheers!
